crates/holochain_types/src/dht_op/facts.rs

```diff
@@ -9,13 +9,28 @@ use holochain_keystore::MetaLairClient;
 /// - The Signature matches the Header
 /// - If the header references an Entry, the Entry will exist and be of the appropriate hash
 /// - If the header does not reference an Entry, the entry will be None
-pub fn valid_dht_op(keystore: MetaLairClient) -> Facts<'static, DhtOp> {
+pub fn valid_dht_op<'a>(
+    keystore: MetaLairClient,
+    author: AgentPubKey,
+    must_be_public: bool,
+) -> Facts<'static, DhtOp> {
     facts![
-        brute("Header type matches Entry existence", |op: &DhtOp| {
-            let has_header = op.header().entry_data().is_some();
-            let has_entry = op.entry().is_some();
-            has_header == has_entry
-        }),
+        brute(
```

Comment:
This feels like test code. Is this whole file behind a test_utils feature?

Reply:
It should be!

Reply:
yep, it is.